Fix API links

drammock opened the pull request:
I noticed a couple of the API links were wrong, so I've corrected them here. In the process I also:

- cleaned some trailing whitespaces
- standardized the fontspec defs in `cheatsheet.tex`

~~WIP because I'm still apparently failing to load all fonts correctly; the text in the "legend placement" section is messed up and is causing the whole thing to spill onto a third page.~~

jklymak commented on cheatsheets.tex:
```diff
@@ -410,7 +415,7 @@
        {}
    \plot{basic-fill.pdf}{\textbf{fill[\_between][x]}( … )}
-       {https://matplotlib.org/api/_as_gen/matplotlib.pyplot.pie.html}
+       {https://matplotlib.org/api/_as_gen/matplotlib.pyplot.fill.html}
```
While we are here, can these all be changed to `https://matplotlib.org/stable/`?  Thanks!

drammock commented:
OK I updated the URLs to use `.../stable/...`.  What I thought was a font problem at first glance was in fact the result of bb2d3d8f7f1976fd7583da92292a3744f316668d not including a recompile of the PDF...  it caused some lines to be too long, which bumped a section to the next column... etc.  I fixed that by cherry-picking the commit from #40, which can now be closed if this gets merged.

rougier commented:
Thanks!
